go/test/endtoend/vtgate/foreignkey/fk_test.go

```diff
@@ -978,20 +978,39 @@ func TestCyclicFks(t *testing.T) {
 
 	// Create a cyclic foreign key constraint.
 	utils.Exec(t, mcmp.VtConn, "alter table fk_t10 add constraint test_cyclic_fks foreign key (col) references fk_t12 (col) on delete cascade on update cascade")
-	defer func() {
-		utils.Exec(t, mcmp.VtConn, "alter table fk_t10 drop foreign key test_cyclic_fks")
-	}()
+
+	matchKsError := func(t *testing.T, keyspace map[string]interface{}) bool {
+		ksErr, fieldPresent := keyspace["error"]
+		if !fieldPresent {
+			return false
+		}
+		errString, isErr := ksErr.(string)
+		if !isErr {
+			return false
+		}
+		// Make sure Vschema has the error for cyclic foreign keys.
+		assert.Contains(t, errString, "VT09019: keyspace 'uks' has cyclic foreign keys")
+		return true
```

Comment:
I feel this should be extracted out into the utils file. The test was more readable before. I think it is good that you have created `WaitForVschemaCondition` function. But we should still keep `WaitForKsError` around and make It internally use `WaitForVschemaCondition`. That way tests will still have the freedom of using the older functions which are more concise to call.